pom.xml

```diff
@@ -97,22 +91,14 @@
         </exclusion>
       </exclusions>
     </dependency>
-
-    <!-- Included for backward compatibility, should be removed once downstream plugins are updated to use okhttp3 -->
-    <dependency>
-      <groupId>com.squareup.okhttp</groupId>
-      <artifactId>okhttp-urlconnection</artifactId>
-      <version>2.7.5</version>
-    </dependency>
```

Comment:
This removal might seem risky and unnecessary, but it is neither. It is necessary because when testing this PR downstream in `github-branch-source-plugin`, I got opaque class loading errors due to the presence of this old library which disappeared when I ripped it out. And the ostensibly reason for its presence is for backward compatibility with downstream plugins that use OkHttp 2.x, but I enumerated all consumers of OkHttp 2.x with `usage-in-plugins`, and while there were a few, none were downstream of `github-api`. From this exercise I conclude that this compatibility layer is no longer necessary.